discovery_engine_core/web-api/src/elastic.rs

```diff
@@ -152,6 +159,64 @@ impl ElasticState {
         Ok(())
     }
 
+    pub(crate) async fn get_document_property(
+        &self,
+        doc_id: &DocumentId,
+        prop_id: &DocumentPropertyId,
+    ) -> Result<Option<DocumentProperty>, Error> {
+        // https://www.elastic.co/guide/en/elasticsearch/reference/8.4/docs-get.html
+        self.query_elastic_search::<DocumentPropertyResponse>(
+            &format!("_source/{doc_id}?_source_includes=properties.{prop_id}"),
```

Comment:
:pushpin: Doesn't this potentially allow injection attacks by choosing problematic document ids and properties.

I would say this is a more generic issue which might be better to fix with a follow up PR but wouldn't it be much better
to use an URL builder or similar to make sure added path segments and properties are appropriately url encoded.

Reply:
good point, i added a [ticket to the board](https://xainag.atlassian.net/browse/ET-3461)